gcc 8.2.0 does not build with isl 0.20

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

Update to `gcc` 8.2.0 after installing `isl` 0.20
- What happened (include command output)

```
==> Upgrading 1 outdated package, with result:
gcc 8.1.0 -> 8.2.0
==> Upgrading gcc 
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.2.0/gcc-8.2.0.tar.xz
Already downloaded: /Users/schoon/Library/Caches/Homebrew/gcc-8.2.0.tar.xz
==> ../configure --build=x86_64-apple-darwin17.7.0 --prefix=/usr/local/Cellar/gcc/8.2.0 --libdir=/usr/local/Cellar/gcc/8.2.0/lib/gcc/8 --enable-languages=c
==> make
Last 15 lines from /Users/schoon/Library/Logs/Homebrew/gcc/02.make:
                 ^
11 errors generated.
make[3]: *** [graphite-isl-ast-to-gimple.o] Error 1
../../gcc/graphite-optimize-isl.c:58:19: error: use of undeclared identifier 'isl_space_dim'
  unsigned dims = isl_space_dim (space, isl_dim_set);
                  ^
../../gcc/graphite-optimize-isl.c:61:3: error: use of undeclared identifier 'isl_space_free'
  isl_space_free (space);
  ^
2 errors generated.
make[3]: *** [graphite-optimize-isl.o] Error 1
rm gcc.pod
make[2]: *** [all-stage1-gcc] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [all] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

```

- What you expected to happen

`gcc` to build without errors
- Step-by-step reproduction instructions (by running `brew install` commands)

```
brew update
brew install --build-from-source gcc
```

@fxcoudert if there is no fix yet in gcc, perhaps we should vendor `isl` 0.19 in for the time being?

--- a/Formula/gcc.rb
+++ b/Formula/gcc.rb
@@ -38,6 +38,10 @@ class Gcc < Formula
     end
   end
 
+  # isl 0.20 compatibility
+  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724
+  patch :DATA
+
   def install
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
@@ -154,3 +158,17 @@ class Gcc < Formula
     assert_equal "Done\n", `./test`
   end
 end
+
+__END__
+diff --git a/gcc/graphite.h b/gcc/graphite.h
+index 4e0e58c..be0a22b 100644
+--- a/gcc/graphite.h
++++ b/gcc/graphite.h
+@@ -37,6 +37,8 @@ along with GCC; see the file COPYING3.  If not see
+ #include <isl/schedule.h>
+ #include <isl/ast_build.h>
+ #include <isl/schedule_node.h>
++#include <isl/id.h>
++#include <isl/space.h>
+
+ typedef struct poly_dr *poly_dr_p;

